File: tests/test_alphacore.py
import unittest
import networkx as nx
import numpy as np
import pandas as pd
from algorithms.alphacore import alphaCore

class TestAlphaCore(unittest.TestCase):
    def setUp(self):
        # Set random seed for reproducibility
        np.random.seed(1)
        
    def test_default_parameters(self):
        """Test with default parameters (200-node graph)"""
        g = nx.erdos_renyi_graph(200, 2/200, directed=True, seed=1)
        # Add edge weights
        for idx, (u, v, w) in enumerate(g.edges(data=True)):
            w['value'] = idx
        
        # Run alphaCore
        result = alphaCore(g)
        
        # Check output structure
        self.assertIsInstance(result, pd.DataFrame)
        self.assertEqual(list(result.columns), ['nodeID', 'alpha', 'batchID'])
        self.assertEqual(len(result), 200)
        self.assertTrue(all((result['alpha'] >= 0) & (result['alpha'] <= 1)))
        self.assertTrue(all(result['batchID'] >= 0))
        
        print(result.head())
        print(result.tail())

if __name__ == '__main__':
    unittest.main()

File: tests/testthat/test_alphaCore.R
# test_alphaCore.R
library(testthat)
library(igraph)
library(data.table)
source("../../algorithms/alphaCore.R")


# Test 1: Default parameters
test_that("Default parameters: ", {
    # Set seed for reproducibility
    set.seed(1)

    g <- erdos.renyi.game(200, 2 / 200, directed = T)
    E(g)$weight <- 1:ecount(g)
    V(g)$name <- paste("v", 1:vcount(g), sep = "")

    # Run alphaCore
    result <- alphaCore(g)

    # Test that result has correct structure
    expect_true(all(c("node", "alpha", "batch") %in% names(result)))
    expect_equal(nrow(result), 200)
    expect_true(all(result$alpha >= 0 & result$alpha <= 1))
    expect_true(all(result$batch >= 0))

    print(head(result, 5))
    print(tail(result, 5))
})

# Test 2: Custom features
test_that("Custom features: ", {
    # Set seed for reproducibility
    set.seed(1)

    g <- erdos.renyi.game(200, 2 / 200, directed = T)
    E(g)$weight <- 1:ecount(g)
    V(g)$name <- paste("v", 1:vcount(g), sep = "")

    result <- alphaCore(g, features = c("indegree", "triangles"))

    # Test that result has correct structure
    expect_true(all(c("node", "alpha", "batch") %in% names(result)))
    expect_equal(nrow(result), 200)
    expect_true(all(result$alpha >= 0 & result$alpha <= 1))
    expect_true(all(result$batch >= 0))

    print(head(result, 6))
    print(tail(result, 5))
})
